cmd/template/cluster/flag.go

```diff
@@ -141,6 +145,10 @@ func (f *flag) Init(cmd *cobra.Command) {
 	cmd.Flags().StringVar(&f.AWS.NetworkVPCCIDR, flagNetworkVPCCidr, "", "CIDR for the VPC.")
 	cmd.Flags().BoolVar(&f.AWS.EKS, flagAWSEKS, false, "Enable AWSEKS. Only available for AWS Release v20.0.0 (CAPA)")
 	cmd.Flags().BoolVar(&f.AWS.ExternalSNAT, flagAWSExternalSNAT, false, "AWS CNI configuration.")
+	cmd.Flags().StringVar(&f.AWS.ClusterType, flagClusterType, "public", "Cluster type to be created (public,proxy-private)")
+	cmd.Flags().StringVar(&f.AWS.HttpsProxy, flagHttpsProxy, "", "Https proxy configuration (required if cluster-type is set to proxy-private)")
```

Comment:
Can we explain what these mean? As a customer I would have no idea.

Reply:
I am open to suggestions because I could not come up more verbose descriptions about them. Besides that, those parameters will be documented `[here](https://docs.giantswarm.io/use-the-api/kubectl-gs/template-cluster/)` I suppose.

Reply:
We should describe whether it's a hostname, a URL, etc.

For the no-proxy flag, I would have no idea what it means, since I expect that "no proxy" means we use no proxy (i.e. nothing has to be configured). So what is that one doing? (Sorry, beginner question.)

Reply:
These are well-known values  `http_proxy` `https_proxy` and `no_proxy` so if the customer has any idea about proxies he will know what to fill in, especially when he has private clusters and wants to set a cluster with proxy, 

if he has no idea he probably need much more reading than what the help could provider

We could mention that it refers to env values like `http_proxy`, `https_proxy`. and `no_proxy`